beacon_node/store/src/database/leveldb_impl.rs

```diff
@@ -0,0 +1,276 @@
+                    metrics::inc_counter_vec(&metrics::DISK_DB_DELETE_COUNT, &[&col]);
+                    let column_key = get_key_for_col(&col, &key);
+                    leveldb_batch.delete(BytesKey::from_vec(column_key));
+                }
+            }
+        }
+        self.db.write(self.write_options().into(), &leveldb_batch)?;
+        Ok(())
+    }
+
+    pub fn begin_rw_transaction(&self) -> MutexGuard<()> {
+        self.transaction_mutex.lock()
+    }
+
+    /// Compact all values in the states and states flag columns.
+    pub fn compact(&self) -> Result<(), Error> {
+        let _timer = metrics::start_timer(&metrics::DISK_DB_COMPACT_TIMES);
+        let endpoints = |column: DBColumn| {
+            (
+                BytesKey::from_vec(get_key_for_col(column.as_str(), Hash256::zero().as_bytes())),
+                BytesKey::from_vec(get_key_for_col(
+                    column.as_str(),
+                    Hash256::repeat_byte(0xff).as_bytes(),
+                )),
+            )
+        };
+
+        for (start_key, end_key) in [
+            endpoints(DBColumn::BeaconStateTemporary),
+            endpoints(DBColumn::BeaconState),
+            endpoints(DBColumn::BeaconStateSummary),
+        ] {
+            self.db.compact(&start_key, &end_key);
+        }
+
+        Ok(())
+    }
+
+    pub fn compact_column(&self, column: DBColumn) -> Result<(), Error> {
+        // Use key-size-agnostic keys [] and 0xff..ff with a minimum of 32 bytes to account for
+        // columns that may change size between sub-databases or schema versions.
+        let start_key = BytesKey::from_vec(get_key_for_col(column.as_str(), &[]));
+        let end_key = BytesKey::from_vec(get_key_for_col(
+            column.as_str(),
+            &vec![0xff; std::cmp::max(column.key_size(), 32)],
+        ));
+        self.db.compact(&start_key, &end_key);
+        Ok(())
+    }
+
+    pub fn iter_column_from<K: Key>(
+        &self,
+        column: DBColumn,
+        from: &[u8],
+        predicate: impl Fn(&[u8], &[u8]) -> bool + 'static,
+    ) -> ColumnIter<K> {
+        let start_key = BytesKey::from_vec(get_key_for_col(column.into(), from));
+
+        let iter = self.db.iter(self.read_options());
+        iter.seek(&start_key);
+
+        Ok(Box::new(
+            iter.take_while(move |(key, _)| {
+                metrics::inc_counter_vec(&metrics::DISK_DB_READ_COUNT, &[column.into()]);
+                let Some(trimmed_key) = key.remove_column_variable(column) else {
+                    return false;
+                };
+                let Some(trimmed_start_key) = start_key.remove_column_variable(column) else {
+                    return false;
+                };
+                key.matches_column(column) && predicate(trimmed_key, trimmed_start_key)
+            })
+            .map(move |(bytes_key, value)| {
+                let key = bytes_key.remove_column_variable(column).ok_or_else(|| {
```

Comment:
You should add the metric here and increase `DISK_DB_READ_BYTES` with value to have the same meaning as the the usage with the per-item gettter

Reply:
[0bab63c](https://github.com/sigp/lighthouse/pull/4718/commits/0bab63cc295850cc7c42e7ecb68afc1da940866a)